wordpress-mobile/WordPress-Android
# Post-Signup Interstitial: Track interstitial screen events

Fixes #10985

For this PR, the following changes were made:

- Four new tracking events were created:
  - `POST_SIGNUP_INTERSTITIAL_SHOWN` 
    - Emmited when the `PostSignupInterstitialActivity` is created.
  - `POST_SIGNUP_INTERSTITIAL_CREATE_NEW_SITE_TAPPED`
    - Emmited when the user taps the **Create WordPress.com Site** button.
  - `POST_SIGNUP_INTERSTITIAL_ADD_SELF_HOSTED_SITE_TAPPED`
    - Emmited when the user taps the **Add Self-Hosted Site** button.
  - `POST_SIGNUP_INTERSTITIAL_DISMISSED`
    - Emmited when the user taps either the **Not Right Now** button or the back button.

- For consistency, the interstitial's back button now has the same behavior as the **Not Right Now** button, meaning the screen will be closed and the user will be redirected to the _Reader_ tab.

- A `PostSignupInterstitialViewModel` has been created alongside some local unit tests.
  - Note: the interstitial screen is fairly simple and a view model isn't really necessary for it to work. The only reason it is there is because it does make local unit tests much easier to write and read. That being said, I thought it would be worth giving up a bit of the implementation's simplicity to achieve simplicity in testing, but I could totally be convinced otherwise. 

## To test

For each of the tests below, you first need to:
1. Clear app data.
2. Start `logcat`.
3. Complete the login flow with an account _that doesn't have any sites_.
4. Confirm that the post-signup interstitial screen is shown.

#### 1. Track interstitial shown
1. On `logcat`, confirm that the `post_signup_interstitial_shown` event was tracked.

#### 2. Track Create New Site button tapped
1. On the post-signup interstitial screen, tap **Create WordPress.com Site**.
2. On `logcat`, confirm that the `post_signup_interstitial_button_tapped` event was tracked with the following properties: `{"button":"create_new_site"}`.

#### 3. Track Add Self-Hosted Site button tapped
1. On the post-signup interstitial screen, tap **Add Self-Hosted Site**.
2. On `logcat`, confirm that the `post_signup_interstitial_button_tapped` event was tracked with the following properties: `{"button":"add_self_hosted_site"}`.

#### 4. Track interstitial dismissed
1. On the post-signup interstitial screen, tap either the **Not Right Now** button or the back button.
2. On `logcat`, confirm that the `post_signup_interstitial_dismissed` event was tracked.

---

PR submission checklist:
